components/kyma-environment-broker/internal/process/deprovisioning/delete_kyma_resource_step.go

```diff
@@ -0,0 +1,61 @@
+package deprovisioning
+
+import (
+	"context"
+	"time"
+
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/process/steps"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/process"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/storage"
+)
+
+const (
+	backoffForK8SOperation = time.Second
+	timeoutForK8sOperation = 10 * time.Second
+)
+
+type DeleteKymaResourceStep struct {
+	operationManager *process.OperationManager
+	kcpClient        client.Client
+}
+
+func NewDeleteKymaResourceStep(operations storage.Operations, kcpClient client.Client) *DeleteKymaResourceStep {
+	return &DeleteKymaResourceStep{
+		operationManager: process.NewOperationManager(operations),
+		kcpClient:        kcpClient,
+	}
+}
+
+func (step *DeleteKymaResourceStep) Name() string {
+	return "Delete_Kyma_Resource"
+}
+
+func (step *DeleteKymaResourceStep) Run(operation internal.Operation, logger logrus.FieldLogger) (internal.Operation, time.Duration, error) {
+	if operation.KymaResourceNamespace == "" {
+		return step.operationManager.OperationFailed(operation, "namespace for Kyma resource not specified", nil, logger)
```

Comment:
confirmed with @piotrmiskiewicz, deprovisioning steps should not set the operation as failed but log and gracefully continue to allow for cluster provisioning

Reply:
changed